File: Frontend/src/models/AttributeDefinition.ts
import {AttributeCategory} from "./AttributeCategory";
import {AttributeDataType} from "./AttributeDataType";
import {OneOfManyOption} from "./OneOfManyOption";

export interface AttributeDefinition {
    id: string;
    name: string;
    dataType: AttributeDataType;
    oneOfManyOptions: OneOfManyOption[] | null;
    attributeCategoryId: string;
    attributeCategory?: AttributeCategory;
}

File: CvManagementSystem/CvManagementSystem.Domain/Models/Attributes/AttributeDefinition.cs
namespace UserService.Domain.Models.Attributes;

public class AttributeDefinition
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public AttributeCategory AttributeCategory { get; set; } = null!;
    public Guid AttributeCategoryId { get; set; }
    public AttributeDataType DataType { get; set; }
}